privacyidea/lib/token.py

```diff
@@ -2857,12 +2864,18 @@ def token_load(token_dict, tokenowner=True, overwrite=False):
     return token
 
 
-def challenge_text_replace(message, user, token_obj):
+def challenge_text_replace(message, user, token_obj, additional_tags: dict = None):
+    # TODO this function should be a token function since most of the info is from that token anyway, optionally pass
+    # TODO environment stuff into that function
     serial = token_obj.token.serial if token_obj.token.serial else None
     tokenowner = user if user else None
     tokentype = token_obj.token.tokentype if token_obj.token.tokentype else ""
     tags = create_tag_dict(serial=serial, tokenowner=tokenowner, tokentype=tokentype)
+    if additional_tags:
+        tags.update(additional_tags)
 
+    if tokentype != "push":
```

Comment:
Is this necessary? If the token type isn't push, there shouldn't be a `presence_answer` tag anyway.

Reply:
Ah, this is in the token loop. Did not see that.